File: cognitum-backend/api/models/game-config.js
import mongoose from "mongoose";

const gameSchema = new mongoose.Schema(
  {
    CurrentLevel: { type: Number, default: 1 },
    MaxLevelReached: { type: Number, default: 1 },
    CountGame: { type: Number, default: 0 },
    MaxScore: { type: Number, default: 0 },
  },
  { _id: false }
);

const gameConfigSchema = new mongoose.Schema({
  userId: { type: mongoose.Schema.Types.ObjectId, required: true, unique: true },
  gamesConfigData: {
    type: Map,
    of: gameSchema,
    default: {},
  },
});

const GameConfig = mongoose.model("GameConfig", gameConfigSchema);

export default GameConfig;

File: cognitum-client/Assets/Scripts/GameConfig/GameConfig.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

/// <summary> Данные игры </summary>
[System.Serializable]
public class GameConfig
{
  public int CurrentLevel = 1;
  public int MaxLevelReached = 1;
  public int CountGame = 0;
  public int MaxScore = 0;
}

/// <summary> Список конфигов игр </summary>
[System.Serializable]
public class GameConfigData
{
  public Dictionary<string, GameConfig> GamesConfigData = new Dictionary<string, GameConfig>();
}
